app/context-menus.js

```diff
@@ -143,6 +155,83 @@ function attachContextMenus ({ window, createWindow }) {
     ]
   }
 
+  function pageGroup (wc) {
+    return [
+      new MenuItem({
+        label: 'Create shortcut',
+        click: async () => {
+          const outputPath = (await dialog.showOpenDialog({
+            properties: ['openDirectory']
+          })).filePaths[0]
+          console.log(outputPath)
+          const appPath = process.argv[0]
+
+          const URL = wc.getURL()
+
+          const shortcutName = wc.getTitle().replace(/[\/|\\:*?"<>| ]/g, '') // Kyran: Normalise into possible file name, maybe we can do this nicer. We get rid of spaces because FS issues.
+
+          const shortcut = {
+            filePath: appPath,
+            outputPath: outputPath,
+            name: shortcutName,
+            comment: 'Agregore Browser',
+            arguments: URL
+          }
+
+          createShortcut = icon => {
+            shortcut.icon = icon
+            // TODO: Kyran: Use Agregore icon if no icon provided.
+            // TODO: Kyran: OSX doesn't have arguments option. See https://github.com/RangerMauve/agregore-browser/pull/53#issuecomment-705654060 for solution.
+            createDesktopShortcut({
+              windows: shortcut,
+              linux: shortcut
+            })
+          }
+
+          let faviconURL
+          try {
+            faviconURL = await wc.executeJavaScript('document.querySelector("link[rel*=\'icon\']").href')
+          } catch {}
+          try {
+            if (!faviconURL) throw 'No favicon'
+            wc.session.on('will-download', (event, item, webContents) => {
```

Comment:
Could this will-download thing be replaced by the same method used to save files?

Reply:
Not sure what you mean by that?

Reply:
If you mean `saveAs` then perhaps we should do it the other way around, replicate the method I've used.

Any advantages to what `saveAs` is doing?